pytorch_lightning/trainer/trainer.py

```diff
@@ -414,6 +414,10 @@ def __init__(
                                        test_percent_check, overfit_pct)
 
         # 16 bit mixed precision training using apex
+        if use_amp:
+            warnings.warn("`use_amp` has been deprecated in favor of `precision` since v0.7.0"
+                          " and will be removed in v0.9.0", DeprecationWarning)
+            precision = 16
```

Comment:
rather add in Deprecated mixin setter like https://github.com/PyTorchLightning/pytorch-lightning/blob/3c2fd560aa4d31b4f48ee225b83361deec53d9c7/pytorch_lightning/trainer/deprecated_api.py#L75-L89